src/main/java/cms/gov/madie/measure/services/TestCaseService.java

```diff
@@ -34,9 +41,27 @@ public TestCase persistTestCase(TestCase testCase, String measureId) {
     return testCase;
   }
 
-  public TestCase updateTestCase(TestCase testCase, String measureId) {
+  public TestCase updateTestCase(TestCase testCase, String measureId, String username) {
     Measure measure = findMeasureById(measureId);
+    List<TestCase> tcs = measure.getTestCases();
+    Instant now = Instant.now();
+    testCase.setLastModifiedAt(now);
+    testCase.setLastModifiedBy(username);
+    int idx =
+        IntStream.range(0, tcs.size())
```

Comment:
This is interesting. Does it provide a benefit over filtering the test case List directly?

```
Optional<TestCase> existingTestCase = measure.getTestCases().stream().filter(tc -> tc.getId().equals((testCase.getId()))).findFirst();
```

Reply:
updated